parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -470,27 +473,50 @@ private void checkAutoCommitIsDisabled(org.apache.kafka.clients.consumer.Consume
+                if (delegate instanceof LegacyKafkaConsumer) {
+                    Field coordinatorField = LegacyKafkaConsumer.class.getDeclaredField("coordinator");
+                    coordinatorField.setAccessible(true);
+                    ConsumerCoordinator coordinator = (ConsumerCoordinator) coordinatorField.get(delegate); //IllegalAccessException
+
+                    if (coordinator == null)
+                        throw new IllegalAccessException("Coordinator for Consumer is null - missing GroupId? Reflection broken?");
+
+                    Field autoCommitEnabledField = coordinator.getClass().getDeclaredField("autoCommitEnabled");
+                    autoCommitEnabledField.setAccessible(true);
+                    isAutoCommitEnabled = (Boolean) autoCommitEnabledField.get(coordinator);
+                } else if (delegate instanceof AsyncKafkaConsumer) {
+                    Field autoCommitEnabledField = AsyncKafkaConsumer.class.getDeclaredField("autoCommitEnabled");
+                    autoCommitEnabledField.setAccessible(true);
+                    isAutoCommitEnabled = (Boolean) autoCommitEnabledField.get(delegate);
+                } else {
+                    log.error("Consumer delegate is neither a LegacyKafkaConsumer nor a AsyncKafkaConsumer - cannot check auto commit is disabled for consumer type: " + consumer.getClass().getName());
+                    isAutoCommitEnabled = false;
+                }
 
-                if (TRUE.equals(isAutoCommitEnabled))
-                    throw new ParallelConsumerException("Consumer auto commit must be disabled, as commits are handled by the library.");
+                if (TRUE.equals(isAutoCommitEnabled)) {
+                    if (options.isSoftAutoCommitDisabledCheck()) {
+                        log.warn("Consumer auto commit must be disabled, as commits are handled by the library.");
+                    } else {
+                        throw new ParallelConsumerException("Consumer auto commit must be disabled, as commits are handled by the library.");
+                    }
+                }
             } else if (consumer instanceof MockConsumer) {
                 log.debug("Detected MockConsumer class which doesn't do auto commits");
             } else {
                 // Probably Mockito
                 log.error("Consumer is neither a KafkaConsumer nor a MockConsumer - cannot check auto commit is disabled for consumer type: " + consumer.getClass().getName());
             }
-        } catch (NoSuchFieldException | IllegalAccessException e) {
-            throw new IllegalStateException("Cannot check auto commit is disabled for consumer type: " + consumer.getClass().getName(), e);
+        } catch (NoSuchFieldException | IllegalAccessException | NullPointerException e) {
+            if (options.isSoftAutoCommitDisabledCheck()) {
+                log.error("Cannot check auto commit is disabled for consumer type: " + consumer.getClass().getName(), e);
```

Comment:
add flag name to make it clear that its only allowed due to the explicit override flag:
```suggestion
                log.error("Cannot check auto commit is disabled for consumer type: {},  Ignoring due to the option flag ignoreReflectiveAccessExceptionsForAutoCommitDisabledCheck=true", consumer.getClass().getName(), e);
```